#159987405 users should be able to create articles

johngorithm opened the pull request:
**What does this PR do?**
Implement create article functionality

**Description of Task to be completed?**
- install `formidable`
- install `cloudinary`
- install `slug`
- install `uuid-random`
- setup `cloudinary`
- write tests for the endpoint
- write a validator middleware

**How should this be manually tested?**
Clone the repository.
Checkout the `ft-Users-should-be-able-to-create-articles-159987405` branch.
Run `npm install` on your local machine.
Run database migrations using `npm run migrate`.
Start the server using `npm start`.

**You can test the following endpoints using postman**
POST `localhost:8000/api/v1/articles`
HEADERS
- authorization -> validjwttoken342klskaus.fdkslie32
- Content-Type -> application/x-www-form-urlencoded

NOTE: 
    Ensure to use `form-data` panel when inputting article data

REQUIRED FIELDS
-    title : `string`
-   description : `string`
-   body : `string`
-   image (optional) : `file`


**What are the relevant pivotal tracker stories?**
#159987405

tomiadebanjo committed:
implement pull request review changes

johngorithm committed:
feat(end-point) : create an article record in the database

- save an article to the database
- upload image to cloudinary
- implement unit testing for the create article endpoint/controller

[Delivered #159987405]

johngorithm committed:
feat(end-point) : create an article record in the database

    - save an article to the database
    - upload image to cloudinary
    - implement unit testing for the create article endpoint/controller
    - validate article require field

[Delivered #159987405]

johngorithm committed:
feat(end-point) : create an article record in the database

        - save an article to the database
        - upload image to cloudinary
        - implement unit testing for the create article endpoint/controller
        - validate article require field

[Delivered #159987405]

johngorithm committed:
feat(end-point) : create an article record in the database

            - save an article to the database
            - upload image to cloudinary
            - implement unit testing for the create article endpoint/controller
            - validate article require field

[Delivered #159987405]

iAmao commented on server/controllers/articleController.js:
```diff
@@ -0,0 +1,82 @@
+          message: 'Article published successfully',
+        });
+      }).catch((error) => {
+        res.status(500).jsend.fail({
```
You should always `return` the response

iAmao commented on server/controllers/articleController.js:
```diff
@@ -0,0 +1,82 @@
+      cloudinary.v2.uploader.upload(
+        temporaryPath, (error, result) => {
+          if (error) {
+            res.status(500).jsend.fail({
```
always `return` the response

iAmao commented on server/models/index.js:
```diff
@@ -5,10 +5,13 @@ import dbConfig from '../config/config';
 
 const basename = path.basename(module.filename);
 const config = dbConfig[process.env.NODE_ENV];
+console.log(config);
```
👍 remove console.log

iAmao reviewed (changes requested):
nice but always return a response

johngorithm committed:
feat(end-point) : create an article record in the database

                - save an article to the database
                - upload image to cloudinary
                - implement unit testing for the create article endpoint/controller
                - validate article require field
		- mock cloudinary API response for testing

[Finished #159987405]

johngorithm committed:
feat(end-point) : create an article record in the database

                    - save an article to the database
                    - upload image to cloudinary
                    - implement unit testing for the create article endpoint/controller
                    - validate article require field
                    - mock cloudinary API response for testing
		    - refactor controller to return response using the `return` keyword

[Finished #159987405]

johngorithm committed:
feat(end-point) : create an article record in the database

        - save an article to the database
        - upload image to cloudinary
        - implement unit testing for the create article endpoint/controller
        - validate article require field
        - mock cloudinary API response for testing
	- refactor controller to return response using the `return` keyword

[Finished #159987405]

johngorithm committed:
Merge branch 'develop' into ft-Users-should-be-able-to-create-articles-159987405

johngorithm committed:
feat(end-point) : create an article record in the database

            - save an article to the database
            - upload image to cloudinary
            - implement unit testing for the create article endpoint/controller
            - validate article require field
            - mock cloudinary API response for testing
            - refactor controller to return response using the `return` keyword

[Finished #159987405]
